File: backend/app/Enums/TimelineAction.php
<?php

declare(strict_types=1);

namespace App\Enums;

enum TimelineAction: string
{
    case CREATED = 'created';
    case ASSIGNED = 'assigned';
    case ASSIGNMENT_UPDATED = 'assignment_updated';
    case STARTED = 'started';
    case HELD = 'held';
    case RESUMED = 'resumed';
    case FINISHED = 'finished';
    case APPROVED = 'approved';
    case CANCELLED = 'cancelled';
    case UPDATED = 'updated';

    public function label(): string
    {
        return match ($this) {
            self::CREATED => __('timeline.actions.created'),
            self::ASSIGNED => __('timeline.actions.assigned'),
            self::ASSIGNMENT_UPDATED => __('timeline.actions.assignment_updated'),
            self::STARTED => __('timeline.actions.started'),
            self::HELD => __('timeline.actions.held'),
            self::RESUMED => __('timeline.actions.resumed'),
            self::FINISHED => __('timeline.actions.finished'),
            self::APPROVED => __('timeline.actions.approved'),
            self::CANCELLED => __('timeline.actions.cancelled'),
            self::UPDATED => __('timeline.actions.updated'),
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::CREATED => 'info',
            self::ASSIGNED => 'primary',
            self::ASSIGNMENT_UPDATED => 'warning',
            self::STARTED => 'success',
            self::HELD => 'warning',
            self::RESUMED => 'primary',
            self::FINISHED => 'success',
            self::APPROVED => 'success',
            self::CANCELLED => 'danger',
            self::UPDATED => 'gray',
        };
    }

    public function icon(): string
    {
        return match ($this) {
            self::CREATED => 'heroicon-o-plus-circle',
            self::ASSIGNED => 'heroicon-o-user-plus',
            self::ASSIGNMENT_UPDATED => 'heroicon-o-pencil-square',
            self::STARTED => 'heroicon-o-play',
            self::HELD => 'heroicon-o-pause-circle',
            self::RESUMED => 'heroicon-o-play-circle',
            self::FINISHED => 'heroicon-o-check-circle',
            self::APPROVED => 'heroicon-o-check-badge',
            self::CANCELLED => 'heroicon-o-x-circle',
            self::UPDATED => 'heroicon-o-pencil',
        };
    }

    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}

File: mobile_new/lib/domain/enums/timeline_action.dart
import 'package:flutter/material.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:easy_localization/easy_localization.dart';

/// Timeline action enum matching Laravel backend
/// Tracks all actions performed on issues and assignments
@JsonEnum(valueField: 'value')
enum TimelineAction {
  @JsonValue('created')
  created('created'),

  @JsonValue('assigned')
  assigned('assigned'),

  @JsonValue('started')
  started('started'),

  @JsonValue('held')
  held('held'),

  @JsonValue('resumed')
  resumed('resumed'),

  @JsonValue('finished')
  finished('finished'),

  @JsonValue('approved')
  approved('approved'),

  @JsonValue('cancelled')
  cancelled('cancelled'),

  @JsonValue('updated')
  updated('updated'),

  @JsonValue('assignment_updated')
  assignmentUpdated('assignment_updated');

  const TimelineAction(this.value);

  /// JSON value for API serialization
  final String value;

  /// Get localized label using easy_localization
  String get label => switch (this) {
    created => 'timeline_action.created'.tr(),
    assigned => 'timeline_action.assigned'.tr(),
    started => 'timeline_action.started'.tr(),
    held => 'timeline_action.held'.tr(),
    resumed => 'timeline_action.resumed'.tr(),
    finished => 'timeline_action.finished'.tr(),
    approved => 'timeline_action.approved'.tr(),
    cancelled => 'timeline_action.cancelled'.tr(),
    updated => 'timeline_action.updated'.tr(),
    assignmentUpdated => 'timeline_action.assignment_updated'.tr(),
  };

  /// Get icon for this action
  IconData get icon => switch (this) {
    created => Icons.add_circle_rounded,
    assigned => Icons.person_add_rounded,
    started => Icons.play_circle_rounded,
    held => Icons.pause_circle_rounded,
    resumed => Icons.play_arrow_rounded,
    finished => Icons.check_circle_rounded,
    approved => Icons.verified_rounded,
    cancelled => Icons.cancel_rounded,
    updated => Icons.edit_rounded,
    assignmentUpdated => Icons.edit_note_rounded,
  };

  /// Get description template
  String getDescription(String performerName) => switch (this) {
    created => '$performerName created this issue',
    assigned => '$performerName assigned this issue',
    started => '$performerName started working',
    held => '$performerName put the work on hold',
    resumed => '$performerName resumed working',
    finished => '$performerName finished the work',
    approved => '$performerName approved the work',
    cancelled => '$performerName cancelled this issue',
    updated => '$performerName edited this issue',
    assignmentUpdated => '$performerName updated the assignment',
  };

  /// Get Arabic description template
  String getDescriptionAr(String performerName) => switch (this) {
    created => 'قام $performerName بإنشاء هذه المشكلة',
    assigned => 'قام $performerName بتعيين هذه المشكلة',
    started => 'بدأ $performerName العمل',
    held => 'قام $performerName بتعليق العمل',
    resumed => 'استأنف $performerName العمل',
    finished => 'أنهى $performerName العمل',
    approved => 'وافق $performerName على العمل',
    cancelled => 'ألغى $performerName هذه المشكلة',
    updated => 'قام $performerName بتعديل هذه المشكلة',
    assignmentUpdated => 'قام $performerName بتحديث التعيين',
  };

  /// Get localized description
  String getLocalizedDescription(String performerName, String locale) =>
      locale == 'ar'
          ? getDescriptionAr(performerName)
          : getDescription(performerName);

  /// Check if this is a positive action
  bool get isPositive => switch (this) {
    created => true,
    assigned => true,
    started => true,
    held => false,
    resumed => true,
    finished => true,
    approved => true,
    cancelled => false,
    updated => true,
    assignmentUpdated => true,
  };

  /// Parse from string value
  static TimelineAction? fromValue(String? value) {
    if (value == null) return null;
    return TimelineAction.values.firstWhere(
      (a) => a.value == value,
      orElse: () => TimelineAction.created,
    );
  }
}
